splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/FunctionIT.java

```diff
@@ -725,5 +726,46 @@ public void testDateTimeToCharVarcharConstants() throws Exception {
             Assert.assertEquals("22018", e.getSQLState());
         }
     }
+
+    @Test
+    public void testCastToCharWithLength() throws Exception {
+        try (ResultSet ignored = methodWatcher.executeQuery("select char(current date, 20)")) {
+            Assert.fail("expect failure since we cannot specify a length with something which is not a char/varchar");
+        } catch (SQLException e) {
+            assertEquals("42846", e.getSQLState());
+            Assert.assertThat(e.getMessage(), containsString("Cannot set a length"));
+        }
+        try (ResultSet ignored = methodWatcher.executeQuery("select char(42, 20)")) {
+            Assert.fail("expect failure since we cannot specify a length with something which is not a char/varchar");
+        } catch (SQLException e) {
+            assertEquals("42846", e.getSQLState());
```

Comment:
why not use SQLState.LANG_INVALID_CAST

Reply:
I'm using a variation of it. Same prefix number, but more comprehensive error message